flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1365,8 +1365,14 @@ IntrinsicLibrary::genIndex(mlir::Type resultType,
   // Call the descriptor-based Index implementation
   auto string = builder.createBox(loc, args[0]);
   auto substring = builder.createBox(loc, args[1]);
+  auto makeRefThenEmbox = [&](mlir::Value b) {
+    auto temp = builder.createTemporary(loc, builder.getI1Type());
```

Comment:
Don't you need a `fir.logical` type rather than `i1` for a logical in a memory interface ? It seems to me that would be safer given that is what the runtime expects/understands.
I guess it works anyway because a while byte or more is anyway always allocated for an `i1`, but using fir.logical would looks sounder.

Reply:
Also had the same thought here. I'm not sure if it makes that much of a difference because this is calling C++ code (which is expecting a `bool` value). Should be easy enough to tweak this.

Reply:
I think the C++ runtime is not treating them as `bool` but expects actual Fortran logical: 
https://github.com/flang-compiler/f18-llvm-project/blob/63f7dbabe2335faf81257aa756845c226d1ea5b7/flang/runtime/character.cpp#L239
So the change you made looks safer to me.